pkg/apis/extensions/types.go

```diff
@@ -332,6 +332,9 @@ type RollingUpdateDeployment struct {
 }
 
 type DeploymentStatus struct {
+	// The generation observed by the deployment controller.
+	ObservedGeneration int `json:"observedGeneration,omitempty"`
```

Comment:
We set Generation in the create hook to 1 so zero essentially means that the controller still hasn't noticed any generation. Which makes the separation between zero and nil reedundant. Also I haven't seen use of pointers in any other status object which makes me believe it's non-idiomatic.